src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -29,30 +41,31 @@ export class ItemTaskAnswerService implements OnDestroy {
     shareReplay(1), // avoid duplicate xhr calls on multiple subscriptions.
   );
 
-  private reloadedAnswerAndState$ = combineLatest([ this.currentAnswer$, this.task$ ]).pipe(
+  private initializedTaskState$ = combineLatest([ this.initialCurrentAnswer$, this.task$ ]).pipe(
     switchMap(([ currentAnswer, task ]) =>
-      (currentAnswer?.state ? task.reloadState(currentAnswer.state).pipe(mapTo({ currentAnswer, task })) : of({ currentAnswer, task }))
+      (currentAnswer?.state ? task.reloadState(currentAnswer.state).pipe(mapTo(undefined)) : of(undefined))
     ),
-    switchMap(({ currentAnswer, task }) =>
+  );
+  private initializedTaskAnswer$ = combineLatest([ this.initialCurrentAnswer$, this.task$ ]).pipe(
+    delayWhen(() => this.initializedTaskState$),
+    switchMap(([ currentAnswer, task ]) =>
       (currentAnswer?.answer ? task.reloadAnswer(currentAnswer.answer).pipe(mapTo(undefined)) : of(undefined))
     ),
-    shareReplay(1),
   );
 
-  private saveAnswerAndStateInterval$ = this.reloadedAnswerAndState$.pipe(
-    switchMap(() => this.task$),
-    switchMap(task => interval(answerAndStateSaveInterval).pipe(mapTo(task))),
-    switchMap(task => forkJoin([ task.getAnswer(), task.getState() ])),
-    distinctUntilChanged(([ answer1, state1 ], [ answer2, state2 ]) => answer1 === answer2 && state1 === state2),
+  private saveAnswerAndStateInterval$ = this.task$.pipe(
+    delayWhen(() => combineLatest([ this.initializedTaskState$, this.initializedTaskAnswer$ ])),
+    repeatLatestWhen(interval(answerAndStateSaveInterval)),
+    switchMap(task => forkJoin({ answer: task.getAnswer(), state: task.getState() })),
+    distinctUntilChanged((a, b) => a.answer === b.answer && a.state === b.state),
     skip(1), // avoid saving an answer right after fetching it
     withLatestFrom(this.config$),
-    switchMap(([ [ answer, state ], { route, attemptId }]) =>
-      this.currentAnswerService.update(route.id, attemptId, { answer, state })
-    ),
+    switchMap(([{ answer, state }, { route, attemptId }]) => this.currentAnswerService.update(route.id, attemptId, { answer, state })),
   );
 
   private subscriptions = [
-    this.reloadedAnswerAndState$.subscribe({ error: err => this.errorSubject.next(err) }),
+    this.initializedTaskAnswer$.subscribe({ error: err => this.errorSubject.next(err) }),
```

Comment:
Doing that, don't you subscribe 3x (so run 3x `task.reloadState`) ? one for each subscription which all depends on initializedTaskAnswer

Reply:
Indeed, I thought the `shareReplay` on `initial(Current)Answer$` was enough, but actually since `switchMap` occurred in the meantime
Fixed by adding a share operator to the initialized* observables

Reply:
👍 
In your test, have you tested failure, to check that the error subject receives what is expected.